include/bluetooth/uuid.h

```diff
@@ -79,6 +79,58 @@ struct bt_uuid_128 {
 #define BT_UUID_32(__u) CONTAINER_OF(__u, struct bt_uuid_32, uuid)
 #define BT_UUID_128(__u) CONTAINER_OF(__u, struct bt_uuid_128, uuid)
 
+
+/**
+ * @brief Encode 128 bit UUID into an array values
+ *
+ * UUID in 128 bit form requires an array creation.
```

Comment:
This is a good justification, but I would rather add this to the commit message instead. Just have an explanation on how you would use this api. With the example given it is very clear.

I would imagine something like this should be enough.
```
 * Helper macro to initialize a 128-bit UUID value from the UUID format.
 * Can be combined with BT_UUID_DECLARE_128 to declare a 128-bit UUID from
 * the readable form of UUIDs.
 *
 * Example for how to declare the UUID `6E400001-B5A3-F393-E0A9-E50E24DCCA9E`
 *
 * @code
 * BT_UUID_128_DECL(BT_UUID_128_ENC(0x6E400001, 0xB5A3, 0xF393, 0xE0A9, 0xE50E24DCCA9E))
 * @endcode
 *
```
```

Reply:
For me it looks reasonable. However I will wait for @dbkinder comment before modifying the documentation.

Reply:
I do like the shorter explanation (together with the @param and @return description).  We don't need the justification and history here in the code (that can go in the commit message).